opp_area_selection/src/selection_artist.cpp

```diff
@@ -0,0 +1,437 @@
+    node["min_cluster_size"] = rhs.min_cluster_size;
+    node["max_cluster_size"] = rhs.max_cluster_size;
+    node["plane_distance_threshold"] = rhs.plane_distance_threshold;
+    node["normal_est_radius"] = rhs.normal_est_radius;
+    node["region_growing_nneighbors"] = rhs.region_growing_nneighbors;
+    node["region_growing_smoothness"] = rhs.region_growing_smoothness;
+    node["region_growing_curvature"] = rhs.region_growing_curvature;
+    return node;
+  }
+
+  static bool decode(const Node& node, opp_area_selection::AreaSelectorParameters& rhs)
+  {
+    if (node.size() != 8)
+    {
+      return false;
+    }
+    rhs.cluster_tolerance = node["cluster_tolerance"].as<decltype (rhs.cluster_tolerance)>();
+    rhs.min_cluster_size = node["min_cluster_size"].as<decltype (rhs.min_cluster_size)>();
+    rhs.max_cluster_size = node["max_cluster_size"].as<decltype (rhs.max_cluster_size)>();
+    rhs.plane_distance_threshold = node["plane_distance_threshold"].as<decltype (rhs.plane_distance_threshold)>();
+    rhs.normal_est_radius = node["normal_est_radius"].as<decltype (rhs.normal_est_radius)>();
+    rhs.region_growing_nneighbors = node["region_growing_nneighbors"].as<decltype (rhs.region_growing_nneighbors)>();
+    rhs.region_growing_smoothness = node["region_growing_smoothness"].as<decltype (rhs.region_growing_smoothness)>();
+    rhs.region_growing_curvature = node["region_growing_curvature"].as<decltype (rhs.region_growing_curvature)>();
+    return true;
+  }
+};
+
+} // end namespace YAML, used to deserialize config file
+
+namespace opp_area_selection
+{
+
+SelectionArtist::SelectionArtist(
+    const ros::NodeHandle& nh,
+    const std::string& world_frame,
+    const std::string& sensor_frame)
+  : nh_(nh)
+  , world_frame_(world_frame)
+  , sensor_frame_(sensor_frame)
+{
+  marker_pub_ = nh_.advertise<visualization_msgs::MarkerArray>(MARKER_ARRAY_TOPIC, 1, false);
+  listener_.reset(new tf::TransformListener (nh_));
+
+  if(!listener_->waitForTransform(sensor_frame_, world_frame_, ros::Time(0), ros::Duration(TF_LOOKUP_TIMEOUT)))
+  {
+    ROS_ERROR("Transform lookup from %s to %s timed out", sensor_frame_.c_str(), world_frame_.c_str());
+    throw std::runtime_error("Transform lookup timed out");
+  }
+
+  clear_roi_points_srv_ = nh_.advertiseService(CLEAR_ROI_POINTS_SERVICE, &SelectionArtist::clearROIPointsCb, this);
+  collect_roi_points_srv_ = nh_.advertiseService(COLLECT_ROI_POINTS_SERVICE, &SelectionArtist::collectROIPointsCb, this);
+
+  // Initialize subscribers and callbacks
+  boost::function<void (const geometry_msgs::PointStampedConstPtr&)> drawn_points_cb;
+  drawn_points_cb = boost::bind(&SelectionArtist::addSelectionPoint, this, _1);
+  drawn_points_sub_ = nh_.subscribe(CLICKED_POINT_TOPIC, 1, drawn_points_cb);
+
+  marker_array_.markers = makeVisual(sensor_frame);
+}
+
+bool SelectionArtist::clearROIPointsCb(
+    std_srvs::TriggerRequest& req,
+    std_srvs::TriggerResponse& res)
+{
+  (void)req;
```

Comment:
Not sure what this does?

Reply:
(void) tells the compiler that a parameter will not be referenced.  I will add a comment to this effect.